src/main/java/net/greghaines/jesque/meta/dao/impl/QueueInfoDAORedisImpl.java

```diff
@@ -225,22 +225,39 @@ private long size(final Jedis jedis, final String queueName) {
     }
 
     /**
-     * Get list of payload from a queue.
+     * Get list of JobPayload from a queue.
      * 
      * @param jedis
      * @param queueName
      * @param jobOffset
      * @param jobCount
      * @return
      */
-    private Collection<String> paylods(final Jedis jedis, final String queueName, final long jobOffset, final long jobCount) {
+    private Collection<JobPayload> getJobPayloads(final Jedis jedis, final String queueName, final long jobOffset, final long jobCount) {
         final String key = key(QUEUE, queueName);
-        final Collection<String> payloads;
-        if (JedisUtils.isDelayedQueue(jedis, key)) { // If delayed queue, use ZRANGE
-            payloads = jedis.zrange(key, jobOffset, jobOffset + jobCount - 1);
+        final Collection<JobPayload> jobPayloads = new ArrayList<>();
+        if (JedisUtils.isDelayedQueue(jedis, key)) { // If delayed queue, use ZRANGEWITHSCORES
+            final Set<Tuple> elements = jedis.zrangeWithScores(key, jobOffset, jobOffset + jobCount - 1);
+            for (final Tuple elementWithScore : elements) {
+                jobPayloads.add(new JobPayload(elementWithScore.getElement(), elementWithScore.getScore()));
+            }
         } else { // Else, use LRANGE
-            payloads = jedis.lrange(key, jobOffset, jobOffset + jobCount - 1);
+            final List<String> elements = jedis.lrange(key, jobOffset, jobOffset + jobCount - 1);
+            for (final String element : elements) {
+                jobPayloads.add(new JobPayload(element, null));
+            }
         }
-        return payloads;
+        return jobPayloads;
     }
+
+    class JobPayload {
+        String element;
+        Double score;
```

Comment:
These properties can both be final since they're only written to in the constructor.